pymess/management/commands/pull_emails_info.py

```diff
@@ -19,7 +23,13 @@ class Command(BaseCommand):
     def handle(self, *args, **options):
         email_sender = get_email_sender()
 
-        messages = EmailMessage.objects.filter(require_pull_info=True).order_by('-sent_at')
+        delay = datetime.timedelta(seconds=settings.EMAIL_PULL_INFO_DELAY_SECONDS)
+        messages = EmailMessage.objects.filter(
+            Q(info_changed_at__isnull=True) | Q(info_changed_at__lt=F('last_webhook_received_at') + delay),
```

Comment:
tady by teoreticky mohlo být 

Q(info_changed_at__isnull=True) | Q(info_changed_at__lt=F('last_webhook_received_at')),

Reply:
nebo spíš mělo ne?

Reply:
delay nastavené na 4 hodiny

1) webhook v čase 12:00
2) spustí se cron v čase 16:30, command běží např. 10 minut a informace se stáhnou až v čase 16:40, tzn. `info_changed_at = 16:40`
3) mezi tím proběhne jiný webhook v čase 16:35, `last_webhook_received_at = 16:35`

V tvém případě se další pull update už neprovede, protože `info_changed_at > last_webhook_received_at`

V mém případě se update ještě provede. Tedy jen v případě, že delay bude větší než délka běhu commanu.

Reply:
Vypadá to, že míchám dva různé delay do jednoho... ale šlo mi v zásadě o to, že:

`info_changed_at__lt` = `last_webhook_received_at` + `delay` + `zpoždění_volání_commandu` + `max_delka_běhu_commandu`

Reply:
V bodě 2) když se pullne info v čase 16:40, tak se tím **mohou** stáhnout data i z webhooku z bodu 3) z času 16:35. Ale jelikož tam také může dojít ke zpoždění (mezi webhookem a dostupností info v API), tak tam musí být ten delay zaveden také.

Reply:
Nestačil by jen ten field `last_webhook_received` s tím že budeš vybírat zprávy které splňují last_webhook_received < now + delay, a při úspěšném stažení informací se `last_webhook_received` nastaví na null?

Reply:
@rubickcz `last_webhook_received_at` by stačil, pokud se shodneme na tom, že nám nevadí, že tuto informaci mažeme.

Ale pak při tom mazání stejně musíme kontrolovat, jestli během provádění commandu nepřišel nový webhhook, protože pak by se `last_webhook_received_at` nesměl smazat. Což pak jen přesouváme logiku z jednoho místa na druhé.

Reply:
Když webhook přijde v době provádění commandu tak to nevadí ne? prostě se zpracuje v dalším spuštění commandu (můžeš ten údaj kontrolovat i těsně před zahájením dotazu do mandrillu). Že se ta informace maže podle mě nevadí - na nic dalšího ji nepotřebujeme. Mně jen přijde to řešení s jedním fieldem jednodušší/intuitivnější, ale nechám to na vás.